src/test/java/steps/StepAssertor.java

```diff
@@ -99,48 +114,59 @@ private static boolean testApplicantEmployees(String json, String responseObject
 
     private static boolean testCategoryCheckNinos(String json, String responseObject, String key, String expected) {
         if (responseObject.startsWith("Category ") && key.contains("nino")) {
-            String jsonPath = "$.categoryChecks..[?(@.calculationType == '" + responseObject + "')].individuals..[?(@.nino == '" + expected + "')]";
+            String jsonPath = getCategoryCheckNinoJsonPath(responseObject, expected);
             JSONArray jsonData = read(json, jsonPath);
             assertTrue(jsonData.size() > 0, String.format("%s %s: Unable to find category check nino %s\nJSON: %s", responseObject, key, expected, json));
             return true;
         }
         return false;
     }
 
-    private static boolean testNinos(String json, String responseObject, String key, String expected) {
-        if (responseObject.toLowerCase().equals("applicant") || responseObject.toLowerCase().equals("partner")) {
-            String jsonPath = "$.individuals..[?(@.nino == '" + expected + "')].nino";
+    private static boolean testCategoryCheck(String json, String responseObject, String key, String expected) {
+        if (responseObject.startsWith("Category ")) {
+            String jsonPath = getCategoryCheckPropertyJsonPath(responseObject, key);
             JSONArray jsonData = read(json, jsonPath);
-            assertTrue(jsonData.size() > 0, String.format("%s %s: Unable to find nino %s\nJSON: %s", responseObject, key, expected, json));
+            String actual = jsonData.get(0).toString().toLowerCase();
+
+            assertTrue(jsonData.size() > 0, String.format("%s %s: Unable to find category check\nJSON: %s", responseObject, key, json));
+            assertTrue(expected.equals(actual), String.format("%s %s: expected %s but actual was %s\nJSON: %s", responseObject, key, expected, actual, json));
             return true;
         }
         return false;
     }
 
-    private static boolean testResponseStatus(String json, String responseObject, String key, String expected) {
-        if (responseObject.toLowerCase().equals("status")) {
-            String jsonPath = "$.status." + key;
-            String actual = read(json, jsonPath).toString().toLowerCase();
-            assertTrue(expected.equals(actual), String.format("Response status %s: expected %s but actual was %s\nJSON: %s", key, expected, actual, json));
-            return true;
+
+    private static List<String> getEmployersList(JSONArray employersWrapper) {
```

Comment:
Could this method be simplified by using a stream? Looks like a mapping and collection.

Reply:
I could have sworn I couldn't get the stream working, hence a good old fashioned iterator.  I was wrong.